cmd/terminus/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	dbsql "database/sql"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+
+	"github.com/treeverse/terminus/pkg/queue_handler"
+	"github.com/treeverse/terminus/pkg/store/sql"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	Execute()
+}
+
+func DieOnErr(err error) {
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func NewSQS() (*sqs.SQS, error) {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("New AWS session: %w", err)
+	}
+	sqs := sqs.New(sess)
+	return sqs, nil
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "terminus",
+	Short: "Terminus monitors and optionally controls quotas for lakeFS users on S3",
+	Long: `Terminus is a service that listens to S3 events in order to track bytes used.
+It may can track S3 resources used by lakeFS installations or users.`,
+}
+
+var runCmd = &cobra.Command{
+	Use:     "run",
+	Short:   "Start the Terminus server",
+	Example: "terminus run --sqs-name=terminus-queue --db-dsn=postgres:///",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		dbDriver, err := cmd.Flags().GetString("db-driver")
+		DieOnErr(err)
+		dbDSN, err := cmd.Flags().GetString("db-dsn")
+		DieOnErr(err)
```

Comment:
This pattern returns multiple times. Maybe we should refactor into designated function:
```go
func GetStringFromFlagsOrDie(flag string) string {
  val, err := cmd.Flags().GetString(flag)
  DieOnErr(err)
  return val
}
```

And then:
```go
dbDriver := GetStringFromFlagsOrDie("db-driver")
dbDsn := GetStringFromFlagsOrDie("db-dsn")
```
etc